### PR TITLE
Issue 30076 analytics ff

### LINKED ISSUE #30076: Create Content Analytics flag And Avoid to run all the Analytics logic if the Analytics App is not config
### Parent Issue

#29711

### User Story

We need to add a flag that allows enabling or disabling the Content Analytics feature as desired. This flag should prevent the Analytics Interceptor from being included and executed when the feature is disabled.

Additionally, if the Analytics app is not configured, none of the Content Analytics processes should run. In this case, the interceptor should still be added and executed, but it should perform no actions.

Finally, we need to ensure that if any exceptions are thrown, the interceptor does not interfere with or block the normal request flow.

### Acceptance Criteria

- Create a Flag to Enabled/disabled the CA.
- Avoid to run any CA code if the Analytics App is not config, Important if you config the App, Content Analytics should start working without restart dotCMS.
- If any Exception is thrown during we executing the CA code the request need to finish normally

### Proposed Objective

Core Features

### Proposed Priority

Please Select

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by jdotcms**
Adding some flag to turn on/off the analytics interceptor

This PR fixes: #30076

### CODE REVIEW

**Review by jcastro-dotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/analytics/track/AnalyticsTrackWebInterceptor.java [hunk 1]
```diff
@@ -1,50 +1,80 @@
...
  */
-public class AnalyticsTrackWebInterceptor  implements WebInterceptor {
+public class AnalyticsTrackWebInterceptor  implements WebInterceptor, EventSubscriber<SystemTableUpdatedKeyEvent> {
 
+    private final static String ANALYTICS_TURNED_ON_KEY = "ANALYTICS_TURNED_ON";
     private final static Map<String, RequestMatcher> requestMatchersMap = new ConcurrentHashMap<>();
+    private final HostWebAPI hostWebAPI;
+    private final AppsAPI appsAPI;
 
     /// private static final String[] DEFAULT_BLACKLISTED_PROPS = new String[]{"^/api/*"};
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/analytics/track/AnalyticsTrackWebInterceptor.java:46]: remove?

File: dotCMS/src/main/java/com/dotcms/analytics/track/AnalyticsTrackWebInterceptor.java [hunk 2]
```diff
@@ -1,50 +1,80 @@
...
     public AnalyticsTrackWebInterceptor() {
 
-        addRequestMatcher(
+        this(WebAPILocator.getHostWebAPI(), APILocator.getAppsAPI(),
+                new WhiteBlackList.Builder()
+                        .addWhitePatterns(Config.getStringArrayProperty("ANALYTICS_WHITELISTED_KEYS",
+                                new String[]{StringPool.BLANK})) // allows everything
+                        .addBlackPatterns(CollectionsUtils.concat(Config.getStringArrayProperty(  // except this
+                                "ANALYTICS_BLACKLISTED_KEYS", new String[]{}), DEFAULT_BLACKLISTED_PROPS)).build(),
+                new AtomicBoolean(Config.getBooleanProperty(ANALYTICS_TURNED_ON_KEY, true)),
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/analytics/track/AnalyticsTrackWebInterceptor.java:59]: Let's rename it to our Feature Flag convention: 
`FEATURE_FLAG_CONTENT_ANALYTICS`

File: dotCMS/src/main/java/com/dotcms/analytics/track/AnalyticsTrackWebInterceptor.java [hunk 3]
```diff
@@ -1,50 +1,79 @@
...
 
 /**
  * Web Interceptor to track analytics
  * @author jsanca
  */
-public class AnalyticsTrackWebInterceptor  implements WebInterceptor {
+public class AnalyticsTrackWebInterceptor  implements WebInterceptor, EventSubscriber<SystemTableUpdatedKeyEvent> {
 
+    private static final String[] DEFAULT_BLACKLISTED_PROPS = new String[]{StringPool.BLANK};
+    private final static String ANALYTICS_TURNED_ON_KEY = "FEATURE_FLAG_CONTENT_ANALYTICS";
```
Comment (dsilvam) [inline dotCMS/src/main/java/com/dotcms/analytics/track/AnalyticsTrackWebInterceptor.java:42]: Don't we need to turn this off in the dotmarketing-properties?

**Review by dsilvam — APPROVED** [review]

File: dotCMS/src/test/java/com/dotcms/analytics/track/AnalyticsTrackWebInterceptorTest.java [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+
+        interceptor.intercept(request, response);
+
+        Assert.assertFalse("The test matcher should be not called, no config set", testMatcher.wasCalled());
+    }
+
+    /**
+     * Method to test: AnalyticsTrackWebInterceptor#intercept
+     * Given Scenario: the feature flag is on and there is config, so the matcher test should be  called
+     * ExpectedResult: The test matcher should be  called
```
Comment (sfreudenthaler) [inline dotCMS/src/test/java/com/dotcms/analytics/track/AnalyticsTrackWebInterceptorTest.java:106]: linting to remove a few extra spaces
```suggestion
     * Given Scenario: the feature flag is on and there is config, so the matcher test should be called
     * ExpectedResult: The test matcher should be called
```

File: dotCMS/src/main/java/com/dotmarketing/filters/InterceptorFilter.java [hunk 5]
```diff
@@ -33,14 +36,17 @@ private void addInterceptors(final FilterConfig config) {
...
 
+        final AnalyticsTrackWebInterceptor analyticsTrackWebInterceptor = new AnalyticsTrackWebInterceptor();
         delegate.add(new MultiPartRequestSecurityWebInterceptor());
         delegate.add(new PreRenderSEOWebInterceptor());
         delegate.add(new EMAWebInterceptor());
         delegate.add(new GraphqlCacheWebInterceptor());
         delegate.add(new ResponseMetaDataWebInterceptor());
         delegate.add(new EventLogWebInterceptor());
         delegate.add(new CurrentVariantWebInterceptor());
-        //delegate.add(new AnalyticsTrackWebInterceptor()); // turn on when needed.
```
Comment (sfreudenthaler) [inline dotCMS/src/main/java/com/dotmarketing/filters/InterceptorFilter.java:43]: thanks past us 😍

File: dotCMS/src/main/resources/dotmarketing-config.properties [hunk 6]
```diff
@@ -857,3 +857,4 @@ CONTENT_EDITOR2_ENABLED=false
 LOCALIZATION_ENHANCEMENTS_ENABLED=false
 
 STARTER_BUILD_VERSION=${starter.deploy.version}
+FEATURE_FLAG_CONTENT_ANALYTICS=false
```
Comment (sfreudenthaler) [inline dotCMS/src/main/resources/dotmarketing-config.properties:860]: For general readability and discoverability, should this be moved up under the `## Telemetry` section ~line 850

**Review by sfreudenthaler — COMMENTED** [review]
tiny nitpicks if this hasn't merged yet

**Pull request merged**